airflow/api_fastapi/parameters.py

```diff
@@ -0,0 +1,165 @@
+        pass
+
+    @abstractmethod
+    def __call__(self, *args: Any, **kwarg: Any) -> BaseParam:
+        pass
+
+    def set_value(self, value: T) -> Self:
+        self.value = value
+        return self
+
+
+class _LimitFilter(BaseParam[int]):
+    """Filter on the limit."""
+
+    def to_orm(self, select: Select) -> Select:
+        if self.value is None:
+            return select
+
+        return select.limit(self.value)
+
+    def __call__(self, limit: int = 100) -> _LimitFilter:
+        self.value = limit
+
+        return self
+
+
+class _OffsetFilter(BaseParam[int]):
+    """Filter on offset."""
+
+    def to_orm(self, select: Select) -> Select:
+        if self.value is None:
+            return select
+        return select.offset(self.value)
+
+    def __call__(self, offset: int = 0) -> _OffsetFilter:
+        self.value = offset
+
+        return self
+
+
+class _PausedFilter(BaseParam[Union[bool, None]]):
+    """Filter on is_paused."""
+
+    def to_orm(self, select: Select) -> Select:
+        if self.value is None:
+            return select
+        return select.where(DagModel.is_paused == self.value)
+
+    def __call__(self, paused: bool | None = Query(default=None)) -> _PausedFilter:
+        return self.set_value(paused)
+
+
+class _OnlyActiveFilter(BaseParam[bool]):
+    """Filter on is_active."""
+
+    def to_orm(self, select: Select) -> Select:
+        if self.value:
+            return select.where(DagModel.is_active == self.value)
+        return select
+
+    def __call__(self, only_active: bool = Query(default=True)) -> _OnlyActiveFilter:
+        return self.set_value(only_active)
+
+
+class _DagIdPatternSearch(BaseParam[Union[str, None]]):
+    """Search on dag_id."""
+
+    def to_orm(self, select: Select) -> Select:
+        if self.value is None:
+            return select
+        return select.where(DagModel.dag_id.ilike(f"%{self.value}"))
+
+    def __call__(self, dag_id_pattern: str | None = Query(default=None)) -> _DagIdPatternSearch:
+        return self.set_value(dag_id_pattern)
+
+
+class SortParam(BaseParam[Union[str]]):
+    """Order result by the attribute."""
+
+    def __init__(self, allowed_attrs: list[str]) -> None:
+        super().__init__()
+        self.allowed_attrs = allowed_attrs
+
+    def to_orm(self, select: Select) -> Select:
+        if self.value is None:
+            return select
+
+        lstriped_orderby = self.value.lstrip("-")
+        if self.allowed_attrs and lstriped_orderby not in self.allowed_attrs:
+            raise HTTPException(
+                400,
+                f"Ordering with '{lstriped_orderby}' is disallowed or "
+                f"the attribute does not exist on the model",
+            )
+        if self.value[0] == "-":
+            return select.order_by(getattr(DagModel, lstriped_orderby).desc())
+        else:
+            return select.order_by(getattr(DagModel, lstriped_orderby).asc())
+        return select
```

Comment:
+1 but then it needs to be:
```suggestion
        return select.order_by(getattr(DagModel, lstriped_orderby).asc())
```

Reply:
Thanks, updated.